MyCryptoHQ/MyCrypto
# Node Refactor (Pt. 3 - Node & Network Cleanup)

### Description

The next in a series of node refactors (Don't worry, I think this is the second to last one.) This one does some cleanup to make nodes and networks more consistent, and makes them a lot easier to manage between static and custom ones. It also adjusts some display, and prevents some errorous situations.

### Changes

* Networks now have an `id` field that works much like `node.id`. It's what should be used for anything logic-based, and is always the key to the in the reducer state object.
* Networks are now unique by chainId. You can't add a custom network whose chainId collides with another network, static or custom.
* Custom network and node actions have been simplified to pass object for add, id for remove
* Network names have been changed to their full name, e.g. `"ETH"` -> `"Ethereum"`
* Network selector now shows `(Testnet)` next to testnet networks.


### Steps to Test

1. Change networks and make sure it all works fine
2. Unlock a web3 wallet and make sure it switches to the web3 node
3. Add a custom node, confirm that works
4. Connect to the custom node, confirm that works
5. Remove a custom node, confirm it's gone
6. Add a custom node with a custom network, confirm that works
8. Connect to the custom node, confirm that works
7. Remove a custom node with a custom network, confirm that both the node and the network are gone


## Screenshots

### No conflicting chainid

![screen shot 2018-04-20 at 1 34 06 pm](https://user-images.githubusercontent.com/649992/39075204-97d664fa-44c2-11e8-91d3-b455a5a09d64.png)

### Testnet badges

![screen shot 2018-04-20 at 4 35 43 pm](https://user-images.githubusercontent.com/649992/39075210-a242ceec-44c2-11e8-8e89-64ea39651bd7.png)
